src/client/options/mod.rs

```diff
@@ -192,6 +192,63 @@ impl fmt::Display for StreamAddress {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[non_exhaustive]
+pub enum ServerApiVersion {
+    Version1,
+}
+
+impl ServerApiVersion {
+    pub fn from_string (version: &str) -> Self {
+        match version {
+            "1" => Self::Version1,
+            _ => panic!("Invalid server API version given")
+        }
+    }
+
+    pub(crate) fn as_str(&self) -> &str {
+        match self {
+            Self::Version1 => "1",
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq, TypedBuilder)]
```

Comment:
`TypedBuilder` can be used to fix one of the issues you mentioned in your top-level comment. It generates a builder pattern for constructing the type and allows you to make fields implemented by default. As it's written right now, the following API would be valid:

```rust
ServerApi::builder()
    .version(some_version)
    .strict(true)
    .deprecation_errors(false)
    .build()
```

However, if you add `#[builder(default)]` above the declarations of the `strict` and `deprecation_errors` fields, then those can be omitted, so the following uses would also be valid:

```rust
ServerApi::builder()
    .version(some_version)
    .strict(true)
    .build()
```
```rust
ServerApi::builder()
    .version(some_version)
    .deprecation_errors(false)
    .build()
```
```rust
ServerApi::builder()
    .version(some_version)
    .build()
```

Reply:
Ah, great to know. Made the change 👍